src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContract.cs

```diff
@@ -328,32 +265,36 @@ internal virtual void WriteRootElement(XmlWriterDelegator writer, XmlDictionaryS
                 writer.WriteStartElement(name, ns);
         }
 
-        internal virtual DataContract GetValidContract(SerializationMode mode)
+        [RequiresUnreferencedCode(DataContract.SerializerTrimmerWarning)]
+        internal virtual DataContract BindGenericParameters(DataContract[] paramContracts, Dictionary<DataContract, DataContract>? boundContracts = null)
         {
             return this;
         }
 
-        internal virtual DataContract GetValidContract()
+        internal virtual DataContract GetValidContract(bool verifyConstructor = false)
         {
```

Comment:
I know this is starting to get out of scope, but these two methods could be combined to a TryGetValidContract which returns the DataContract via an out parameter and returns bool. Just a thought I had when understanding the code.